### ISSUE TITLE
ioredis client not supported

### ISSUE THREAD

**Issue by dmamolina** [PatrickJS/redis-dataloader#3]
First of all thank you for this integration. I just wanted to drop a line since I found a problem when trying to integrate this library with the [ioredis](https://github.com/luin/ioredis)  client.

I didn't find any issues with:
* [redis](https://www.npmjs.com/package/redis)
* [redis-sentinel](https://www.npmjs.com/package/redis-sentinel)

The response returned by ioredis when calling the `multi.exec` method is different from the response of other clients. Instead of returning and array of results, it returns and array of arrays.

redis response example:

```
[ 'OK', 1, '{"id":133....']
```

ioredis response example:

```
[ [null, 'OK'], [null, 1], [null, '{"id":133....']]
```

Since the rSetAndGet is trying to parse the last element of that array it will fail in the ioredis case:

```js
const rSetAndGet = (keySpace, key, rawVal, opt) => toString(rawVal, opt)
.then(val => Q.Promise((resolve, reject) => {
    const fullKey = makeKey(keySpace, key);
    const multi = redis.multi();
    multi.set(fullKey, val);
    if(opt.expire) {
        multi.expire(fullKey, opt.expire);
    }
    multi.get(fullKey);
    multi.exec((err, replies) => err ?
        reject(err) : parse(_.last(replies), opt).then(resolve)
    );
}));
```

**Comment by scf4** [comment]
@dmamolina I was in the same boat as you.

I tried to add support to redis-dataloader but found the code to be unnecessarily complex and hard to follow and ended up rewriting the whole thing.

https://github.com/scf4/ioredis-dataloader

Very untested, use at own risk etc

**Comment by DubFriend** [comment]
Hey sorry I haven't responded to this earlier, I've been pretty swamped at work lately.

Thank you for the detailed report, this definitely saved me some time.

I added support for `ioredis`. See https://github.com/DubFriend/redis-dataloader/commit/c8a96ce75b564b4e9867dd2f884cb1f0dd4324e7
